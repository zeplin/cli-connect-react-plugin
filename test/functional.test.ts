import Plugin from "../src";

describe("Connected Components React Plugin - Functional", () => {
    test("FunctionalComponents.jsx snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process(
            {
                path: "test/samples/jsx-functional/FunctionalComponent.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("FunctionalComponentWithChildren.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-functional/FunctionalComponentWithChildren.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("MultiExportFunctionalComponentWithChildren.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-functional/MultiExportFunctionalComponentWithChildren.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
