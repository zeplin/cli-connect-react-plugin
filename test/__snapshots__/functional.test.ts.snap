// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components React Plugin - Functional FunctionalComponentWithChildren.jsx snippet creation 1`] = `
Object {
  "description": "",
  "lang": "jsx",
  "snippet": "<MyComponent>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - Functional FunctionalComponents.jsx snippet creation 1`] = `
Object {
  "description": "",
  "lang": "jsx",
  "snippet": "<MyComponent />",
}
`;

exports[`Connected Components React Plugin - Functional MultiExportFunctionalComponentWithChildren.jsx snippet creation 1`] = `
Object {
  "description": "Component 1 description. Only this one should be shown",
  "lang": "jsx",
  "snippet": "<MyComponent1>
  {children}
</MyComponent1>

<MyComponent2
  someParameter={PropTypes.string.isRequired}>
  {children}
</MyComponent2>",
}
`;
